torch/_decomp/decompositions_for_rng.py

```diff
@@ -0,0 +1,292 @@
+from collections import defaultdict
+from typing import Callable, Dict
+
+import torch
+import torch._decomp as decomp
+from torch._ops import OpOverload
+from torch._subclasses.fake_tensor import disable_fake_tensor_mode_tracing
+
+aten = torch.ops.aten
+
+rng_decompositions: Dict[str, Dict[OpOverload, Callable]] = defaultdict(dict)
+offset_calculator = {}
```

Comment:
Why is this concept needed?

Reply:
Offsets will change from operator to operator. So, along with decomp per op, we also have offset_calculator per op.

Reply:
Agreed. Changed it.